src/bot/file_receiver.py

```diff
@@ -0,0 +1,122 @@
+
+    async def classify_and_handle_message(self, message: types.Message):
+        try:
+            message_type = self.determine_message_type(message)
+
+            if message_type == "link":
+                await self.handle_link(message)
+            elif message_type == "document":
+                await self.handle_document(message)
+            elif message_type == "video":
+                await self.handle_video(message)
+            else:
+                await self.handle_unknown_content(message)
+
+        except Exception as e:
+            logging.error(f"Error occurred: {e}")
+            await message.answer("Error processing the message.")
+
+    async def handle_link(self, message: types.Message):
+        # TODO Handle link-specific logic
+        metadata = self.create_metadata(message)
+        self.print_metadata(metadata)
+        await message.answer("This is a link")
+
+    async def handle_document(self, message: types.Message):
+        # TODO Handle document-specific logic, including size check
+        if message.document.file_size > self.DOCUMENT_SIZE_LIMIT:
+            await message.answer("Document size exceeds 100 MB limit.")
+            return
+        metadata = self.create_metadata(message)
+        self.print_metadata(metadata)
+        # TODO Further processing (e.g., storage)
+
+    async def handle_video(self, message: types.Message):
+        # TODO Handle video-specific logic, including size check
+        if message.video.file_size > self.VIDEO_SIZE_LIMIT:
+            await message.answer("Video size exceeds 500 MB limit.")
+            return
+        metadata = self.create_metadata(message)
+        self.print_metadata(metadata)
+        # TODO Further processing (e.g., storage)
+
+    async def handle_unknown_content(self, message: types.Message):
+        # TODO Handle unknown content type logic
+        await message.answer("Unknown content type")
+
+    def create_metadata(self, message: types.Message):
+        sender = message.from_user
+        sender_id = sender.id if sender else None
+
+        if sender and sender.username:
+            sender_username = f"@{sender.username}"
+        else:
+            sender_username = "No username"
+
+        if sender:
+            sender_name = sender.first_name
+            if sender.last_name:
+                sender_name += f" {sender.last_name}"
+        else:
+            sender_name = "No name"
+
+        message_type = self.determine_message_type(message)
+        timestamp = message.date.isoformat()
+        file_size = self.get_file_size(message)
+
+        metadata = {
+            "sender_id": sender_id,
+            "sender_username": sender_username,
+            "sender_name": sender_name,
+            "message_type": message_type,
+            "timestamp": timestamp,
+            "file_size in bytes": file_size,
+        }
+
+        return metadata
+
+    def print_metadata(self, metadata):
```

Comment:
testing method?

Reply:
The method is for testing and useful during this review phase. I plan to remove it before the final merge, after resolving all other discussions.